projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	c := http.Client{Timeout: time.Duration(1) * time.Second}
+
+	resp, err := c.Get("http://localhost:8080")
+
+	if err != nil {
+		fmt.Fprint(os.Stderr, "Something went wrong. Cannot complete request at the moment\n")
+		return
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 429:
+		retryTime := resp.Header.Get("Retry-After")
+		formattedRetryTime, err := parseRetryValue(retryTime)
+
+		if err != nil {
+			// cannot determine how long to sleep for, retry once after 1 sec delay. Reason - The server response shows it is safe to retry. However we might not want keep at it for long - hence the choice
+
+			sleep(1)
+			resp, err = retry(1, c)
+
+			if err != nil {
+				// retry still unsuccessful and server does not respond
+				fmt.Fprint(os.Stderr, "Something went wrong. Cannot complete request at the moment\n")
+				resp.Body.Close()
+				return
+			}
+
+		} else if formattedRetryTime > 5 {
+			// sleep time is long
+			fmt.Fprint(os.Stderr, "Server busy. Cannot retrieve weather\n")
+			resp.Body.Close()
+			return
+		} else {
+			sleep(formattedRetryTime)
+			resp, err = retry(3, c) // chose 3 as max number of retries
+
+			if err != nil {
+				// retry still unsuccessful and server does not respond
+				fmt.Fprint(os.Stderr, "Something went wrong. Cannot complete request at the moment\n")
+				resp.Body.Close()
+				return
+			}
+
+		}
+	}
+
+	// for other responses - print to stdout
+	body, _ := io.ReadAll(resp.Body)
```

Comment:
Any time you ignore an error, I'd recommend adding a comment above explaining why the error is safe to ignore so the next person reading (or debugging!) your code understands the thought process you went through and can benefit from the time you spent reasoning about it.